lib/portlayer/exec/exec.go

```diff
@@ -171,3 +184,35 @@ func publishContainerEvent(id string, created time.Time, eventType string) {
 	VCHConfig.EventManager.Publish(ce)
 
 }
+
+func WaitForContainerStop(ctx context.Context, id string) error {
+	defer trace.End(trace.Begin(id))
+
+	listen := make(chan interface{})
+	defer close(listen)
+
+	watch := func(ce events.Event) {
+		event := ce.String()
+		if ce.Reference() == id {
+			switch event {
+			case events.ContainerStopped,
+				events.ContainerPoweredOff:
+				listen <- event
+			}
+		}
+	}
+
+	sub := fmt.Sprintf("%s:%s(%d)", id, "watcher", &watch)
+	topic := events.NewEventType(events.ContainerEvent{}).Topic()
+	VCHConfig.EventManager.Subscribe(topic, sub, watch)
+	defer VCHConfig.EventManager.Unsubscribe(topic, sub)
+
+	// wait for the event to be pushed on the channel or
+	// the context to be complete
+	select {
+	case <-listen:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
```

Comment:
Let's wrap that error here or in the caller, otherwise we have another possible CDE error without any context.